src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -265,15 +288,24 @@ private CallableStatement getCallableStatementHandle(CallableHandles request,
             String catalog) throws SQLServerException {
         CallableStatement CS = null;
         HandleAssociation hassoc = handleMap.get(request);
-        if (null == hassoc || null == hassoc.databaseName || !hassoc.databaseName.equals(catalog)) {
-            CS = request.prepare(connection);
-            hassoc = new HandleAssociation(catalog, CS);
-            HandleAssociation previous = handleMap.put(request, hassoc);
-            if (null != previous) {
-                previous.close();
+        try {
```

Comment:
I'm having a hard time understanding the logic change behind HandleAssociation. Can you explain to me why we need another map inside HandleAssociation?